### DISCUSSION TITLE
The_problem_of_interface_control_points

### DISCUSSION THREAD

**Discussion by wt150203** [SPECFEM/specfem3d#1639] [SPECFEM3D General]
The following file is in "specfem3d-master\EXAMPLES\seismoacoustic_bishop2022\DATA_FIG4\meshfem3D_files". I'm curious about "spacing between points which must be constant". How could I determine the correct value of it? Looking forward to hearing from you, I will be grateful ！
# number of interfaces
2
#
# We describe each interface below, structured as a 2D-grid, with several parameters : 
# number of points along XI and ETA, minimal XI ETA coordinates 
# and spacing between points which must be constant.
# Then the records contain the Z coordinates of the NXI x NETA points.
#
# interface number 1
 .true. 2 2 652000d0 3493000d0 50.0d0  50.0d0
 interface1.dat
# interface number 5
 .true. 2 2 652000d0 3493000d0 50.0d0  50.0d0
 interface2.dat

# for each layer, we give the number of spectral elements in the vertical direction
# layer number 1 (bottom layer)
 18
# layer number 5 (top layer)
 45

**Comment by homnath** [comment]
The spacing is generally determined by the interface topography. For example, if you have a flat interface, you can use two points in each direction with spacing equal to the dimensions in each direction. On the other hand, if the interface has a complex topography, the spacing should be small enough to represent the realistic topography. In this case, the number of points will be more.
